core/src/main/java/cz/xtf/core/bm/BuildManager.java

```diff
@@ -16,6 +17,10 @@ public BuildManager(OpenShift openShift) {
         if (openShift.getProject(openShift.getNamespace()) == null) {
             openShift.createProjectRequest();
             openShift.waiters().isProjectReady().waitFor();
+            // Adding a label can be only done via 'namespace'. It cannot be set via 'project' API.
+            // Thus we do this separately. Also, updating namespaces seems to be restricted only to Admin users.
+            OpenShifts.admin().namespaces().withName(openShift.getNamespace()).edit().editMetadata().addToLabels(
```

Comment:
Question I have is whether use of `admin` is required. Isn't `master` user enough?

Reply:
Using a regular `master` user led me into this error when I used this build in one of our project that utilizes this XTF:
```
Failure executing: PATCH at: https://api.nmsp-prne-wll3.eapqe.psi.redhat.com:6443/api/v1/namespaces/jstourac-namespace. Message: namespaces "jstourac-namespace" is forbidden: User "xpaasqe" cannot patch resource "namespaces" in API group "" in the namespace "jstourac-namespace". Received status: Status(apiVersion=v1, code=403, details=StatusDetails(causes=[], group=null, kind=namespaces, name=jstourac-namespace, retryAfterSeconds=null, uid=null, additionalProperties={}), kind=Status, message=namespaces "jstourac-namespace" is forbidden: User "xpaasqe" cannot patch resource "namespaces" in API group "" in the namespace "jstourac-namespace", metadata=ListMeta(_continue=null, remainingItemCount=null, resourceVersion=null, selfLink=null, additionalProperties={}), reason=Forbidden, status=Failure, additionalProperties={}).
org.apache.maven.surefire.booter.SurefireBooterForkException: There was an error in the forked process
```

Also I can see similar error message, when I try manually via `oc` command:
```
$ oc label namespace test "label1=true"
Error from server (Forbidden): namespaces "test" is forbidden: User "xpaasqe" cannot patch resource "namespaces" in API group "" in the namespace "test"
```

Via `admin` user this operation succeeds:
```
$ oc label namespace test "label1=true" && oc delete project test
namespace/test labeled
```

Based on this, I suppose that `admin` user is required.

Reply:
Ok, I'm not sure under which user namespace is created but if it's `admin` then I guess it's ok.

Reply:
Both, `BuildManager` and `ProjectCreator` create a project as `master` user, not `admin`. I don't think it is important, `admin` is only needed to update labels on namespace.

Reply:
There are cases we run xtf on clusters where we do not have admin rights. E.g. s390x testing. We run tests which does not need admin permissions there. 

But with this change introduced that will fail in such cases.

Reply:
good point, I think in that case we should only log warning that it isn't possible label namespace?

Reply:
If there really isn't way how to avoid xtf dependency on `admin()` rights than this is for sure better than fail.

So for environments where cluster_admin is not available it will stay basically same.

These cleaning job. I need to be admin I suppose to create them anyway? And if admin of shared cluster is willing to create such cleaning job, it can be still made like we have now e.g. `if (creator=xpaasuser && createdDate > 7days)` ?

I think this xtf optional dependency on permission `patch resource "namespaces" in API group "" in the namespace "???"` should be mentioned somewhere. I will let @mnovak1 identify what is the proper location. Is it README?

Reply:
Okay, I tried to put an if statement in latest `wip` commit so it wouldn't fail in case admin is not available. TBH - this doesn't seem to work yet, since e.g. if I add wrong admin password, then I get `java.lang.ExceptionInInitializerError`. Looks like current code is not able to cope with admin account not available? I'll check more into this later, but this change is to check whether you feel more comfortable with such solution.

Reply:
Also, I played a bit with the role binding yesterday and I found out that when I add a `cluster-admin` role-binding to the `xpaasqe` user on particular namespace, then he is then able to edit namespace lables... But I believe that we won't be able to provide such permissions, right? At least dynamically this would be quite tricky as such permission change has to be done by `admin` user anyway... Not sure whether we are able to have our OCP instance pre-configured somehow to incorporate that permission automatically.

Reply:
Yes, I am ok with if condition.

Our clusters is not problem, we could configure whatever we agree on. Problem is IBM cluster, which is maanged by Ops and they won't allow admin permission (somehow understandable). 

`ExceptionInInitializerError` interesting. Where is it from? when you use xpaasqe user in place of admin is it same?

Reply:
When I use wrong password (on purpose) for admin account :) I was just testing what happens. I'm not quite familiar with how this code is supposed to behave. As such, yes, this needs to be modified.